Document/0x05c-Reverse-Engineering-and-Tampering.md

```diff
@@ -290,6 +290,91 @@ For white-box source code testing, you'll need a setup similar to the developer'
+
+With Ghidra, strings can be obtained by simply loading the dex file into the tool and by selecting **Window -> Defined strings** in the menu.
+
+> Loading APK file directly to Ghidra is buggy, thus it is recommended to extract dex file by unzipping the APK file and using it in Ghidra.
+
+<img src="Images/Chapters/0x05c/ghidra_dex_strings.png" width="500px"/>
+
+
+Dextra is a CLI based tool, and provides multiple features to explore dex files. All the strings in a dex file can be dumped using Dextra using the following command:
+
+```bash
+dextra -S classes.dex
+```
+
+The output from Dextra can be manipulated using standard Linux commands, for example, using `grep` to search for certain keywords.
+
+It is important to know, the list of strings obtained using the above tools can be very big, as it also includes the various class and package names used in the application. Going through the complete list, specially for big binaries, can be very cumbersome. Thus, it is recommended to start with keyword based searching and go through the list only when keyword search does not work. Some keywords which can be a good starting point are - password, key, and secret.
+
+
+###### Native
+
+For extracting strings from the native code used in an Android application, we can either use Ghidra (GUI based) or `strings` utility command available on \*nix.
+
+Use of Ghidra for extracting strings for a native binary has already been discussed in "[Using Disassemblers and Decompilers](0x04c-tampering-and-reverse-engineering#ghidra "Using Disassembler and Decompilers")" section.
+
+`strings` is a CLI based tool and using it is straightforward, as demonstrated below for a native library:
+
+```bash
+strings libnative-lib.so
+...
+Java_sg_vantagepoint_helloworldjni_MainActivity_stringFromJNI
+...
+Hello from C++
+...
+```
+
+##### Cross References
+
+###### Java
+Java cross references is supported by many tools, including Ghidra and [jadx](https://github.com/skylot/jadx "jadx"). With Ghidra cross references can be obtained by right clicking on the desired function in the symbol tree and selecting **Show References to**, as shown in the image below.
```

Comment:
Please put in a newline after the headings. Also in the next paragraphs

Reply:
Sorry on this one. I thought my linter took care of it. Fixed now.